enrollment-server-onboarding-provider-microblink/src/main/java/com/wultra/app/onboardingserver/provider/microblink/MicroblinkDocumentVerificationProvider.java

```diff
@@ -244,7 +243,18 @@ public List<String> parseRejectionReasons(DocumentResultEntity docResult) {
 
     @Override
     public VerificationSdkInfo initVerificationSdk(OwnerId id, Map<String, String> initAttributes) {
-        return new VerificationSdkInfo();
+        logger.info("Initializing Microblink SDK for activationId {}", id.getActivationId());
+
+        final var mobilePlatform = initAttributes.getOrDefault("mobilePlatform", null);
+        logger.info("Requested sdk license key for platform {}", mobilePlatform);
+
+        final var sdkInfo = new VerificationSdkInfo();
+
+        Optional.ofNullable(mobilePlatform)
+                .map(platform -> mobileSdkLicenseKeyByPlatform.getOrDefault(platform, null))
+                .ifPresent(licenseKey -> sdkInfo.getAttributes().put("licenseKey", licenseKey));
```

Comment:
Ok, it is working and consumers are expected to make side-effect but have you considered more functional way?

```suggestion
        return Optional.ofNullable(mobilePlatform)
        .map(platform -> mobileSdkLicenseKeyByPlatform.get(platform))
        .map(licenseKey -> {
            return new VerificationSdkInfo(Map.of("licenseKey", licenseKey);
        })
        .orElseGet(VerificationSdkInfo::new);
```

Reply:
Yes, I like it more